src/System.Management.Automation/FormatAndOutput/DefaultFormatters/PowerShellCore_format_ps1xml.cs

```diff
@@ -789,30 +789,212 @@ private static IEnumerable<FormatViewDefinition> ViewsOf_System_Management_Autom
+                                            $offsetLength = $myinv.PositionMessage.split('+')[$highlightLine].Trim().Length
+                                            $line = $line.Insert($myinv.OffsetInLine - 1 + $offsetLength, $resetColor)
+                                            $line = $line.Insert($myinv.OffsetInLine - 1, $errorColor)
+                                            $posmsg += ""${accentColor}${lineWhitespace}$($myinv.ScriptLineNumber) ${verticalBar} ${resetcolor}${line}`n""
+                                            $offsetWhitespace = ' ' * ($myinv.OffsetInLine - 1)
+                                            $prefix = ""${accentColor}${headerWhitespace}     ${verticalBar} ${errorColor}""
+                                            $message = ""${prefix}${offsetWhitespace}^ ""
+                                        }
+
+                                        if (! $_.ErrorDetails -or ! $_.ErrorDetails.Message) {
+                                            # we use `n instead of $newline here because that's what is in the message
+                                            if ($_.CategoryInfo.Category -eq 'ParserError' -and $_.Exception.Message.Contains(""~`n"")) {
+                                                # need to parse out the relevant part of the pre-rendered positionmessage
+                                                $message += $_.Exception.Message.split(""~`n"")[1].split(""${newline}${newline}"")[0]
+                                            }
+                                            else {
+                                                $message += $_.Exception.Message
+                                            }
+                                        }
+                                        else {
+                                            $message += $_.ErrorDetails.Message
+                                        }
+
+                                        # if rendering line information, break up the message if it's wider than the console
+                                        if ($myinv -and $myinv.ScriptName -or $_.CategoryInfo.Category -eq 'ParserError') {
+                                            $prefixLength = Get-RawStringLength -string $prefix
+                                            $prefixVtLength = $prefix.Length - $prefixLength
+
+                                            # replace newlines in message so it lines up correct
+                                            $message = $message.Replace($newline, ' ')
+                                            if ([Console]::WindowWidth -gt 0 -and ($message.Length - $prefixVTLength) -gt [Console]::WindowWidth) {
+                                                $sb = [Text.StringBuilder]::new()
+                                                $substring = Get-TruncatedString -string $message -length ([Console]::WindowWidth + $prefixVTLength)
```

Comment:
I think there are conditions where `[console]::windowwidth` is 0 (`docker run --rm powershell pwsh -c [console]::windowwidth` is one). Not sure if this matters

Reply:
line 930 covers this